Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1284,6 +1284,25 @@ static HTMLElement* topmostPopoverAncestor(Element& newPopover)
     return topmostAncestor.get();
 }
 
+// https://html.spec.whatwg.org/#popover-focusing-steps
+static void runPopoverFocusingSteps(HTMLElement& popover)
+{
+    RefPtr control = popover.findFocusDelegate();
+
+    if (!control)
+        return;
+
+    if (control->isFocusable())
```

Comment:
I am not sure we should be checking for `isFocusable`.

Was reading through the focusing steps and step 1 says.

```
If new focus target is not a [focusable area](https://html.spec.whatwg.org/#focusable-area), then set new focus target to the result of [getting the focusable area](https://html.spec.whatwg.org/#get-the-focusable-area) for new focus target, given focus trigger if it was passed.
```

Reply:
@rniwa Do we implement "getting the focusable area" somewhere? I'm trying to implement https://html.spec.whatwg.org/#popover-focusing-steps

Reply:
Not really. The closest we have is `Element::focusAppearanceUpdateTarget()`.

Reply:
I removed the check for `isFocusable()`, we should fix `runFocusingStepsForAutofocus` separately to use `focusAppearanceUpdateTarget` if needed.